homeassistant/components/person/__init__.py

```diff
@@ -569,3 +577,44 @@ def _get_latest(prev: State | None, curr: State):
     if prev is None or curr.last_updated > prev.last_updated:
         return curr
     return prev
+
+
+class ListPersonsView(HomeAssistantView):
+    """List all persons if request is made from a local network."""
+
+    requires_auth = False
+    url = "/api/person/list"
+    name = "api:person:list"
+
+    async def get(self, request: web.Request) -> web.Response:
+        """Return a list of persons if request comes from a local IP."""
+        try:
+            remote_address = ip_address(request.remote)  # type: ignore[arg-type]
+        except ValueError:
+            return self.json_message(
+                message="Invalid remote IP",
+                status_code=HTTPStatus.BAD_REQUEST,
+                message_code="invalid_remote_ip",
+            )
+
+        hass: HomeAssistant = request.app["hass"]
+        if is_cloud_connection(hass) or not is_local(remote_address):
+            return self.json_message(
+                message="Not local",
+                status_code=HTTPStatus.BAD_REQUEST,
+                message_code="not_local",
+            )
```

Comment:
Should we consider to make this a decorator?

Reply:
Currently, this code is used only once in the entire code base. We could make a decorator out of it when we implement it a second time, or are there other features pending that should be available only on a local IP?

Reply:
Let's convert it into a decorator as soon as we have more use-cases.